data-products/src/main/scala/org/sunbird/analytics/exhaust/collection/ResponseExhaustJob.scala

```diff
@@ -29,23 +29,24 @@ object ResponseExhaustJob extends optional.Application with BaseCollectionExhaus
       "questionscore" -> "Question Score", "questionmaxscore" -> "Question Max Score", "questionoption" -> "Question Options", "questionresponse" -> "Question Response")
 
   override def processBatch(userEnrolmentDF: DataFrame, collectionBatch: CollectionBatch)(implicit spark: SparkSession, fc: FrameworkContext, config: JobConfig): DataFrame = {
-    val assessmentDF = getAssessmentDF(collectionBatch).persist();
+    val assessmentDF = getAssessmentDF(userEnrolmentDF, collectionBatch).persist();
     persistedDF.append(assessmentDF);
     val contentIds = assessmentDF.select("content_id").dropDuplicates().collect().map(f => f.get(0));
     val contentDF = searchContent(Map("request" -> Map("filters" -> Map("identifier" -> contentIds)))).withColumnRenamed("collectionName", "contentname").select("identifier", "contentname");
-    val reportDF = assessmentDF.join(contentDF, assessmentDF("content_id") === contentDF("identifier"), "left_outer").join(userEnrolmentDF, Seq("courseid", "batchid", "userid"), "left_outer").drop("identifier").select(filterColumns.head, filterColumns.tail: _*);
+    val reportDF = assessmentDF.join(contentDF, assessmentDF("content_id") === contentDF("identifier"), "left_outer").drop("identifier").select(filterColumns.head, filterColumns.tail: _*);
     organizeDF(reportDF, columnMapping, columnsOrder);
   }
   
   override def unpersistDFs() {
     persistedDF.foreach(f => f.unpersist(true))
   }
   
-  def getAssessmentDF(batch: CollectionBatch)(implicit spark: SparkSession, fc: FrameworkContext, config: JobConfig): DataFrame = {
-    loadData(assessmentAggDBSettings, cassandraFormat, new StructType()).where(col("course_id") === batch.collectionId && col("batch_id") === batch.batchId)
-      .withColumnRenamed("user_id", "userid")
-      .withColumnRenamed("batch_id", "batchid")
-      .withColumnRenamed("course_id", "courseid")
+  def getAssessmentDF(userEnrolmentDF: DataFrame, batch: CollectionBatch)(implicit spark: SparkSession, fc: FrameworkContext, config: JobConfig): DataFrame = {
+    val userEnrolmentDataDF = getEnrolmentData(userEnrolmentDF, batch)
```

Comment:
Why is this code required? this is getting filtered in BaseCollectionExhaust anyway

Reply:
Removed this code.